Fix for #6661. Module Manager/Hathor. Reset filters Position, Language and Type when filter Client changes.

bertmert opened the pull request:
Issue https://github.com/joomla/joomla-cms/issues/6661

Test:
- Current staging.
- Extensions > Module Manager
- Select "Site" as Client to see frontend modules.
- Select Position "atomic-bottomleft".
- Select Language: "All"
- Select now "Administrator" as Client to see backend modules.
- Message "There are no modules matching your query"
- See Position filter field: "- Select Position-". That's confusing.
- See Language filter field: Still "All"
- Click cross next to Search filter to reload page. All OK afterwards.

LOG  OUT, LOG IN! to clear user session. Then apply patch. Or vice versa.

After patch:
- Filters for Position and Type are resetted correctly when Client changed. Also in the background.
- Message "There are no modules matching your query" only if other filters like State or Language don't match. 
- You see modules of newly selected client immediately. There's no need to reload page.
- Test also in template Hathor!

bertmert committed:
Clear #filter_position and #filter_module when client_id changes

bertmert committed:
Update default.php

crommie commented:
Tested according to instructions. All filters are reset, except Language filter, shouldn't that be reset as well?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6673">issues.joomla.org/joomla-cms/6673</a>.</sub>

larapetersen commented:
@test Could recreate problem, patch works as described. I agree that it looks nicer  if language filter is reset as well. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6673">issues.joomla.org/joomla-cms/6673</a>.</sub>

gaastras commented:
@test Patch is working correctly.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6673">issues.joomla.org/joomla-cms/6673</a>.</sub>

bertmert committed:
Update default.php

Removed JavaScript. Due to https://github.com/joomla/joomla-cms/pull/6682 not needed anymore.

bertmert commented:
Due to RTC PR https://github.com/joomla/joomla-cms/pull/6682 (that fixes this issue similarly) I've removed changes of administrator/components/com_modules/views/modules/tmpl/default.php here.
Just fix for template Hathor remaining here.

brianteeman commented:
Setting to RTC - thanks for testing<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6673">issues.joomla.org/joomla-cms/6673</a>.</sub>

roland-d commented:
I have removed the RTC label for now, because I do agree with @crommie and @larapetersen that the language filter should be reset as well. If I set it to English for example, changing to Administrator also gives me no results. Resetting all search filters, will give you all modules and you can filter from there on.

roland-d commented:
@bertmert Why close this PR? I think it is a good change, just need to add the language reset as well.

bertmert committed:
added reset for #filter_language

bertmert commented:
Added reset for #filter_language. I coudn't test it (crashed my test environment somehow).

infograf768 commented:
This does not work here and I do not understand how anyone could test it correctly. 
This is for Hathor, not Isis.
<b>The issue is not related to this PR. It happens here without this change</b>

I never can get the Site modules when using Hathor (as we have to click on Submit) if the client was set to Administrator in Isis before I switched to Hathor

If I set the filter on Site and let all filters set to Select ... or choose something in any filter, when I click on Submit, the Client changes to Administrator as it was set to Administrator in Isis before switching to Hathor.

If, in the contrary, I set in Isis the client to Site, then I never can display the Administrator client modules in Hathor.

To be sure of this test. I set the template to Hathor, then quitted joomla and deleted the cookie, then logged again: I never could get the administrator modules to display as, per default, the client is site.

Better correct that bug first imho.

<b>PS: Related, needs a small PR</b>: `#filter_language` is missing for Isis
https://github.com/joomla/joomla-cms/blob/staging/administrator/components/com_modules/layouts/joomla/searchtools/default/bar.php#L18

infograf768 commented:
I guess the bug came from:
https://github.com/joomla/joomla-cms/pull/6682

bertmert commented:
@brianteeman  @roland-d  @infograf768 I've closed this PR because of https://github.com/joomla/joomla-cms/pull/6682#issuecomment-100203377 Some code parts of 6682 are related to this issue here. I don't find a way at the moment to solve all conflicts. I'm too confused about Hathor specifics,,,

Bakual commented:
PR #6682 is reverted.
